File: class6 exercices/Hour_client.py
import sys
import rospy
from class_6.srv import *

def hour_client(x, y):

	    rospy.wait_for_service('hour')
	    try:
		Hour = rospy.ServiceProxy('hour', hour)
		resp1 = Hour(x)
		return resp1.hour
	    except rospy.ServiceException, e:
		print "Service call failed: %s"%e


def usage():
    return "%s [x y]"%sys.argv[0]

if __name__ == "__main__":

    while True :
	if len(sys.argv) == 3:
		x = int(sys.argv[1])
		y = int(sys.argv[2])
	else:
		print usage()
		sys.exit(1)
	print "hour : %s"%(hour_client(x, y))
    	rospy.sleep(1);

File: classe6/Hour_client.cpp
#include "ros/ros.h" /*	*/
#include "class_6/hour.h" /* On récupère le service créé */
#include <cstdlib> /* */

int main(int argc, char **argv)
{
  ros::init(argc, argv, "hour_client");
  if (argc != 3)
  {
    ROS_INFO("usage: hour client");
    return 1;
  }

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<class_6::hour>("hour");
  class_6::hour srv;
  srv.request.test = 1;

  while(1){

	  if (client.call(srv))
	  {
	    ROS_INFO("Hour: %ld", srv.response.hour);
	  }
	  else
	  {
	    ROS_ERROR("Failed to call service hour");
	    return 1;
	  }
	  ros::Duration(1).sleep();
  }

  return 0;
}
